### PR TITLE
Release notes for beta-20160714

### PR DESCRIPTION
**Pull request by bdarnell**
cockroachdb/cockroach#7804

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/453)

### CODE REVIEW

**Comment by jseldess** [comment]
:lgtm: I'll add links and doc updates once this is merged.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/453#-:-KMdjNi-zTZwHrZxvG4w:395610063)*

**Comment by jseldess** [comment]
Actually, just realized that we're missing the new column family heuristics. 

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/453#-:-KMdjsFchJaP72hEiVyw:-1334462703)*

**Comment by paperstreet** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[beta-20160714.md, line 11 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdk-FdMenDorJYhtS6:-KMdk-FdMenDorJYhtS7:1134018108) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L11)):*
> ```Markdown
> ### Upgrade Notes
> 
> - This release cannot be run concurrently with older beta releases. Please stop all nodes running older releases before restarting any node with this version.
> ```

We should explictly mention that once you start up this version on your data, you can't use any versions older than beta-20160629

---

*[beta-20160714.md, line 16 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdl_L3g26Yj15cU66i:-KMdl_L3g26Yj15cU66j:-1988901892) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L16)):*
> ```Markdown
> ### New Features
> 
> - CockroachDB can now be built on FreeBSD. [#7545](https://github.com/cockroachdb/cockroach/pull/7545)
> ```

Column families are now used by default, so tables created with this release should be much faster

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/453)*

**Comment by bdarnell** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[beta-20160714.md, line 11 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdk-FdMenDorJYhtS6:-KMe-IikkXKSIQPqZ_4V:441083763) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L11)):*
<details><summary><i>Previously, paperstreet (Daniel Harrison) wrote…</i></summary>
> We should explictly mention that once you start up this version on your data, you can't use any versions older than beta-20160629
</details>
Our support for downgrading is generally not very good and we haven't been calling it out in the release notes. Have we gotten good enough about this that it's safe to downgrade unless we otherwise call it out?

---

*[beta-20160714.md, line 16 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdl_L3g26Yj15cU66i:-KMe-UZqpd54Xd_Qjv2H:1505755500) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L16)):*
<details><summary><i>Previously, paperstreet (Daniel Harrison) wrote…</i></summary>
> Column families are now used by default, so tables created with this release should be much faster
</details>
What should we recommend for existing tables? Is there an ALTER TABLE command  that will upgrade to the new format or should we recommend dropping and recreating tables if possible?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/453)*

**Comment by paperstreet** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[beta-20160714.md, line 11 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdk-FdMenDorJYhtS6:-KMe0n_MWOSAWjtuZi-2:504203045) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L11)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> Our support for downgrading is generally not very good and we haven't been calling it out in the release notes. Have we gotten good enough about this that it's safe to downgrade unless we otherwise call it out?
</details>
Yeah, I think we were downgradeable from when we fixed the decimal issues right after beta until beta-20160629.

We have a (fairly incomplete) acceptance test that checks this `TestDockerReadWriteBidirectionalReferenceVersion`

---

*[beta-20160714.md, line 16 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/453#-KMdl_L3g26Yj15cU66i:-KMe1Y5wh-ZVCzbRYadr:1943692528) ([raw file](https://github.com/cockroachdb/docs/blob/e59cc6f731601803e1197a5099978eb964f483e1/beta-20160714.md#L16)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> What should we recommend for existing tables? Is there an ALTER TABLE command  that will upgrade to the new format or should we recommend dropping and recreating tables if possible?
</details>
Upgrades are not available for this, so you have to drop and recreate if you want it.

Or create duplicates of all the non-pk columns with the family config you want, copy data over, and drop the old ones, but that seems crazy at this point

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/453)*

File: beta-20160714.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Raft commands for queries that have been abandoned by the client are no longer retried internally. [#7605](https://github.com/cockroachdb/cockroach/pull/7605)
+- Table leases are now released when connections are closed, allowing schema changes to proceed without waiting for leases to expire. [#7661](https://github.com/cockroachdb/cockroach/pull/7661)
+
+### Bug Fixes
+
+- The command-line SQL client is now smarter about semicolons and `\` commands inside string literals. [#7510](https://github.com/cockroachdb/cockroach/pull/7510)
+- `LIMIT` is now applied correctly on queries that use `JOIN`. [#7546](https://github.com/cockroachdb/cockroach/pull/7546)
+- The "CGo Calls" graph in the web UI is now displayed as a rate instead of a cumulative total. [#7597](https://github.com/cockroachdb/cockroach/pull/7597)
+- The `cockroach quit` command now waits for the server to stop before returning. [#7603](https://github.com/cockroachdb/cockroach/pull/7603)
+- Buildng CockroachDB in a Docker container on Linux with `build/builder.sh` no longer requires the container and host toolchains to be the same. [#7626](https://github.com/cockroachdb/cockroach/pull/7626)
```
Comment (petermattis) [inline beta-20160714.md]: `s/Buildng/Building/g`

Comment (jseldess) [inline beta-20160714.md]: Fixed in https://github.com/cockroachdb/docs/pull/455

**Pull request merged**